bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BXMLSequence.java

```diff
@@ -462,4 +462,13 @@ public boolean hasNext() {
             return cursor < value.sequence.size();
         }
     }
+
+    @Override
+    public void removeChildren(String qname) {
+        if (sequence.size() != 1) {
```

Comment:
Shouldn't this check be < 1 instead? Also what if the size is zero?

Reply:
size != 1 will work.

size > 1 means its a sequence, which means its not an element, hence cannot remove a children.
size is zero means its an empty sequence. In that case again we can't remove..